src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,170 @@
+import { Component, Element, h, Host, Listen, Prop, VNode } from "@stencil/core";
+
+@Component({
+  tag: "calcite-inline-editable",
+  shadow: true,
+  styleUrl: "calcite-inline-editable.scss"
+})
+export class CalciteInlineEditable {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el!: HTMLCalciteInlineEditableElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Props
+  //
+  //--------------------------------------------------------------------------
+
+  @Prop({ mutable: true, reflect: true }) editingEnabled = false;
+
+  @Prop({ mutable: true, reflect: true }) loading = false;
+
+  @Prop() onConfirmChanges: () => Promise<any>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Getters/Setters
+  //
+  //--------------------------------------------------------------------------
+
+  get hasAction(): boolean {
+    return !!this.onConfirmChanges;
+  }
+
+  get shouldShowActions(): boolean {
+    return this.editingEnabled && this.hasAction;
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  componentWillLoad() {
+    this.inputElement = this.el.querySelector("calcite-input");
+    this.inputElement.uneditable = true;
+  }
+
+  render(): VNode {
+    return (
+      <Host scale={this.inputElement.scale}>
```

Comment:
Would be good to turn `scale` into a formal prop with `@Prop() scale: string = this.inputElement.scale` that way storybook knobs can be used.

Reply:
WIll that work? `inputElement` isn't initialized until `componentWillLoad`. Was also hoping to avoid exposing a prop we probably don't want consumers to set. My logic being we probably don't want consumers to use different scales across the inline-editable and input components.

Reply:
You should be able to initialize it like:
```
private inputElement: HTMLCalciteInputElement = this.el.querySelector("calcite-input");
```
without the need to do it in a lifecycle method.

Reply:
Also it feels kinda weird to set a prop that relies on a child.  If that child wasn't present or the scale was set incorrectly, the styles wouldn't apply as intended because the @Prop is not defined with a default value in `inline-editable`.  I'd say `inline-editable` should have a declared scale prop with a default value as a fallback and should be used to control the scale of the slotted input for design consistency, even if the child's scale might differ from the parent's.

@bstifle @macandcheese  curious your thoughts on whether the `scale` prop should be controllable and documented for `inline-editable` and if it should override the slotted input's scale?

Reply:
I think it should look for a scale on wrapping label if present. If not it can fan back to its own. Input "looks up" to a wrapping label in this way, so ideally seeing setting scale on a wrapping label affects any child input, input message, or now editable content components.

Agree it's a little weird to "look down" vs having a scale prop itself.

I think we can do what input does here - accept a prop and also look up to the wrapping parent label to set.

Reply:
It really feels superfluous to have to set a prop on the inline-editable component. To avoid that I'm incorporating your feedback re: looking up the tree while also allowing it to grab the props off the wrapped input. I felt this was necessary because I don't think inline editable inputs will always be wrapped with labels. For instance, we'll be relying on the placeholder do that that work in our app. It then seems strange to set some props on the child input and some on the inline editable component. This new approach should get us the best of both worlds.

Reply:
I think for accessibility, labels are a must and should be preferred over placeholders. e.g. https://www.deque.com/blog/accessible-forms-the-problem-with-placeholders/

Also, what determines which `scale` prop wins when there's both a scale set on the outer label as well as the input?  Sounds like that will be confusing both to the maintainers as well as the users who might be expecting both input and inline-editable to look up for their scale prop.

Plus, if input already looks up to a wrapping label for scale, it could also look up for a scale from inline-editable.  Perhaps that might make more sense.  🤷